sentinelhub/api/batch/utils.py

```diff
@@ -86,39 +214,28 @@ def monitor_batch_job(
     return tiles_per_status
 
 
-def monitor_batch_analysis(
-    batch_request: BatchRequestType, config: Optional[SHConfig], sleep_time: int = _DEFAULT_ANALYSIS_SLEEP_TIME
-) -> BatchRequest:
-    """A utility function that is waiting until analysis phase of a batch job finishes and regularly checks its status.
-    In case analysis phase failed it raises an error at the end.
+def _monitor_batch_statistical_execution(
+    batch_request: BatchStatisticalRequest, config: Optional[SHConfig], sleep_time: int
+) -> None:
+    batch_client = SentinelHubBatchStatistical(config=config)
 
-    :param batch_request: An object with information about a batch request. Alternatively, it could only be a batch
-        request id or a payload.
-    :param config: A configuration object with required parameters `sh_client_id`, `sh_client_secret`, and
-        `sh_auth_base_url` which is used for authentication and `sh_base_url` which defines the service deployment
-        where Batch API will be called.
-    :param sleep_time: Number of seconds between consecutive status updates during analysis phase.
-    :return: Batch request info
-    """
-    if sleep_time < _MIN_ANALYSIS_SLEEP_TIME:
-        raise ValueError(
-            f"To avoid making too many service requests please set analysis sleep time >={_MIN_ANALYSIS_SLEEP_TIME}"
-        )
+    def current_completion_percentage() -> float:
+        """Fetches the current completion percentage"""
+        return batch_client.get_status(batch_request)["completionPercentage"]
 
-    batch_client = SentinelHubBatch(config=config)
+    progress = current_completion_percentage()
 
-    batch_request = batch_client.get_request(batch_request)
-    while batch_request.status in [BatchRequestStatus.CREATED, BatchRequestStatus.ANALYSING]:
-        LOGGER.info("Batch job has a status %s, sleeping for %d seconds", batch_request.status.value, sleep_time)
-        time.sleep(sleep_time)
-        batch_request = batch_client.get_request(batch_request)
+    with tqdm(total=100, initial=progress, desc="Completion percentage") as progress_bar:
+        while progress < 100:
+            time.sleep(sleep_time)
```

Comment:
This time could probably be by default around 5-10 seconds I guess. I actually recommend checking with Marjan about this.